api/test/integration/listing.e2e-spec.ts

```diff
@@ -2468,6 +2468,121 @@ describe('Listing Controller Tests', () => {
     });
   });
 
+  describe('dynamic required fields validation', () => {
+    let customJurisdiction;
+    let defaultJurisdiction;
+
+    beforeAll(async () => {
+      // Create a jurisdiction with custom required fields
+      customJurisdiction = await prisma.jurisdictions.create({
+        data: jurisdictionFactory(randomName(), {
+          requiredListingFields: ['name', 'listingsBuildingAddress'],
+        }),
+      });
+
+      // Create a jurisdiction without specified required fields (should use default strict validation)
+      defaultJurisdiction = await prisma.jurisdictions.create({
+        data: jurisdictionFactory(randomName()),
+      });
+    });
+
+    it('should allow saving active listing with requiredListingFields declared in jurisdiction', async () => {
+      const listingData = {
+        name: 'Detroit Minimal Listing',
+        status: ListingsStatusEnum.active,
+        listingsBuildingAddress: addressFactory() as AddressCreate,
+        jurisdictions: {
+          id: customJurisdiction.id,
+        },
+      };
+
+      const res = await request(app.getHttpServer())
+        .post('/listings')
+        .set({ passkey: process.env.API_PASS_KEY || '' })
+        .send(listingData)
+        .set('Cookie', adminAccessToken)
+        .expect(201);
+
+      expect(res.body.name).toEqual(listingData.name);
+      expect(res.body.status).toEqual(ListingsStatusEnum.active);
+    });
+
+    it('should fail saving active listing without requiredListingFields declared in jurisdictions', async () => {
+      const incompleteData = {
+        name: 'Detroit Minimal Listing',
+        status: ListingsStatusEnum.active,
+        jurisdictions: {
+          id: customJurisdiction.id,
+        },
+      };
+
+      // Should fail due to missing required fields from requiredListingFields
+      await request(app.getHttpServer())
```

Comment:
suggestion: can we add a validation that the correct error message is returned? Same goes for the other tests that expect a 400